// react-bootstrap-month-selector/dist/RangePopover.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _reactBootstrap = require("react-bootstrap");

var _moment = require("moment");

var _moment2 = _interopRequireDefault(_moment);

var _MonthSelector = require("./MonthSelector");

var _MonthSelector2 = _interopRequireDefault(_MonthSelector);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var monthPropType = _react.PropTypes.shape({
  month: _react.PropTypes.number.isRequired,
  year: _react.PropTypes.number.isRequired
});

var styles = {
  footer: {
    padding: 0,
    border: "none"
  }
};

var RangePopover = function (_React$Component) {
  _inherits(RangePopover, _React$Component);

  function RangePopover(props) {
    _classCallCheck(this, RangePopover);

    var _this = _possibleConstructorReturn(this, (RangePopover.__proto__ || Object.getPrototypeOf(RangePopover)).call(this));

    _this.onSetStart = _this.onSetStart.bind(_this);
    _this.onSetEnd = _this.onSetEnd.bind(_this);
    _this.onSave = _this.onSave.bind(_this);
    _this.onClose = _this.onClose.bind(_this);
    _this.onChange = _this.onChange.bind(_this);

    _this.state = {
      start: { year: 2012, month: 1 },
      end: { year: 2012, month: 2 }
    };
    return _this;
  }

  _createClass(RangePopover, [{
    key: "onSetStart",
    value: function onSetStart(start) {
      this.setState({ start: start }, this.onChange);
    }
  }, {
    key: "onSetEnd",
    value: function onSetEnd(end) {
      this.setState({ end: end }, this.onChange);
    }
  }, {
    key: "onClose",
    value: function onClose() {
      this.props.onClose && this.props.onClose();
    }
  }, {
    key: "onChange",
    value: function onChange() {
      var _state = this.state,
          start = _state.start,
          end = _state.end;

      this.props.onChange && this.props.onChange(this.state);
    }
  }, {
    key: "onSave",
    value: function onSave() {
      var _state2 = this.state,
          start = _state2.start,
          end = _state2.end;

      this.props.onSave && this.props.onSave(this.state);
    }
  }, {
    key: "render",
    value: function render() {
      var _state3 = this.state,
          start = _state3.start,
          end = _state3.end;

      var _props = this.props,
          onSave = _props.onSave,
          onClose = _props.onClose,
          onChange = _props.onChange,
          _s = _props.start,
          _e = _props.end,
          showButtons = _props.showButtons,
          props = _objectWithoutProperties(_props, ["onSave", "onClose", "onChange", "start", "end", "showButtons"]);

      return _react2.default.createElement(
        _reactBootstrap.Popover,
        props,
        _react2.default.createElement(
          "div",
          { className: "form-group" },
          _react2.default.createElement(_MonthSelector2.default, {
            month: start.month,
            year: start.year,
            onSelect: this.onSetStart
          })
        ),
        _react2.default.createElement(
          "div",
          { className: "form-group" },
          _react2.default.createElement(_MonthSelector2.default, {
            month: end.month,
            year: end.year,
            onSelect: this.onSetEnd
          })
        ),
        showButtons ? _react2.default.createElement(
          "div",
          { className: "modal-footer", style: styles.footer },
          _react2.default.createElement(
            _reactBootstrap.Button,
            { onClick: this.onClose },
            "Close"
          ),
          _react2.default.createElement(
            _reactBootstrap.Button,
            { bsStyle: "primary", onClick: this.onSave },
            "Save"
          )
        ) : null
      );
    }
  }]);

  return RangePopover;
}(_react2.default.Component);

RangePopover.propTypes = {
  onSave: _react.PropTypes.func,
  onClose: _react.PropTypes.func,
  onChange: _react.PropTypes.func,
  showButtons: _react.PropTypes.func.bool,
  start: monthPropType.isRequired,
  end: monthPropType.isRequired
};
RangePopover.defaultProps = {
  showButtons: true
};
exports.default = RangePopover;
